content/servicemesh/routing.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Intelligent Routing"
+date: 2018-11-13T21:49:32+09:00
+weight: 50
+draft: false
+---
+
+### Intelligent Routing
+
+> Deploying a microservice-based application in an Istio service mesh allows one to externally control service monitoring and tracing, request (version) routing, resiliency testing, security and policy enforcement, etc., In a consistent way across the services, for the application as a whole.
+
+Before you can use Istio to control the Bookinfo version routing, you need to define the available versions, called <span style="color:orange">*`subsets`*</span>, in destination rules
+
+```
+kubectl apply -f samples/bookinfo/networking/destination-rule-all.yaml
+
+kubectl get destinationrules -o yaml
+```
+
+To route to one version only, you apply virtual services that set the default version for the microservices. In this case, the virtual services will route all traffic to <span style="color:orange">**reviews:v1**</span> of each microservice.
+
+```
+kubectl apply -f samples/bookinfo/networking/virtual-service-all-v1.yaml
+
+kubectl get virtualservices reviews -o yaml
+```
+
+> Iterate reloading the page and check out review section calls only version of reviews v1 all the time
+
+Change the route configuration so that all traffic from a specific user is routed to a specific service version. In this case, all traffic from a user named <span style="color:orange">*Jason*</span> will be routed to the service <span style="color:orange">**reviews:v2**</span>.
+
+```
+kubectl apply -f samples/bookinfo/networking/virtual-service-reviews-test-v2.yaml
+
+kubectl get virtualservices reviews -o yaml
+```
+
+> Login to a user as jason. You will only see reviews:v2 all the time. Others will see reviews:v1
```

Comment:
Click **Sign in** on top right corner and login using 'jason' as user name with blank password.

Reply:
replaced all the "login to a user as jason" to yours